### PR TITLE
InsteonPLM binding improvements and bug fixes

### PR DESCRIPTION
**Pull request by berndpfrommer**
This is consolidated and sanitized commit of work done by Bernd & Daniel Pfrommer and Rob Nielsen:

- added support for Insteon Hub 2014 (Daniel)
- fixed race condition in writer/reader reply status (Bernd & Rob)
- added support for power meter (Rob Nielsen)
- better handling for multiple outstanding queries (Bernd)
- more detailed debug messages for command and message handlers (Bernd)
- higher layer support for extracting multiple data fields in single message handler (Bernd)
- improved suppression of duplicate messages for dimmers (Rob)
- support for delaying outgoing messages (Bernd & Rob)
- replaced GreedyDispatcher with SimpleDispatcher (Bernd)

Testing: 24h production testing on 40+ devices (Rob),  24h production testing on 110+ devices (Bernd).

### CODE REVIEW

**Comment by teichsta** [comment]
Hi @berndpfrommer, thanks for this contribution! Anything mission critical to be cherry-picked to 1.6.2 (i know you've send just one commit - so i can take it all or nothing)? Best, Thomas E.-E.

**Comment by berndpfrommer** [comment]
First I don't think there is anything too mission critical.

Certainly I don't understand the build process, and don't know what the
state of the 1.6.2 binding is. I certainly never tested the changes against
1.6.2, so I'd say no, this should only go into 1.7.

**Comment by teichsta** [comment]
ok. thanks!

- sent from a mobile device -

> Am 17.01.2015 um 04:00 schrieb Bernd Pfrommer <notifications@github.com>:
> 
> First I don't think there is anything too mission critical. 
> 
> Certainly I don't understand the build process, and don't know what the 
> state of the 1.6.2 binding is. I certainly never tested the changes against 
> 1.6.2, so I'd say no, this should only go into 1.7. 
> 
>

**Comment by berndpfrommer** [comment]
Hold off on this one, I got a deadlock after 48h. I think I understand what the issue is. Will test fix and will add it to this PR.

**Comment by berndpfrommer** [comment]
After 48h of testing it looks as though the binding no longer hangs. Please consider for merge, thanks!

**Comment by robnielsen** [comment]
I've also been testing with no hangs.

**Comment by robnielsen** [comment]
This pull request also fixes the exception in https://groups.google.com/forum/#!topic/openhab/ME8r8vFslZE

**Comment by robnielsen** [comment]
Can we get this approved?

**Comment by berndpfrommer** [comment]
@teichsta, please let me know if there are any outstanding issues that prevent this PR to be merged.

**Comment by teichsta** [comment]
for now it's "only" a little bit of spare time :) I'll let you know! Thanks, Thomas E.-E.

File: bundles/binding/org.openhab.binding.insteonplm/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -29,3 +29,5 @@ Bundle-DocURL: http://www.openhab.org
 Bundle-RequiredExecutionEnvironment: JavaSE-1.6
 Service-Component: OSGI-INF/activebinding.xml, OSGI-INF/genericbindingprovider.xml
 Bundle-ClassPath: .
+Require-Bundle: org.apache.httpcomponents.httpclient;bundle-version="4.1.3",
+ org.apache.httpcomponents.httpcore;bundle-version="4.1.4"
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.insteonplm/META-INF/MANIFEST.MF]: Require-Bundle should be avoided wherever possible. Please use Import-Package instead

**Comment by teichsta** [comment]
no more review comments … let me know, when you've incorporated the Feedback.

Thanks, Thomas E.-E.

**Comment by berndpfrommer** [comment]
I cannot figure out how to do get the import done.
If I add to Import-Package:

 org.apache.http,
 org.apache.http.auth,
 org.apache.http.client,
 org.apache.http.client.methods

I get this error when I try mvn clean install:

[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.19.0:compile (default-compile) on project org.openhab.binding.insteonplm: Compilation failure: Compilation failure:
[ERROR] /home/pfrommer/workspace/openhab/bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/driver/hub/HubIOStream.java:[22]
[ERROR] import org.apache.http.impl.client.DefaultHttpClient;
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The type DefaultHttpClient is not accessible due to restriction on classpath entry /home/pfrommer/.m2/repository/p2/osgi/bundle/org.apache.httpcomponents.httpclient/4.1.3.v2012092

What access restriction is that? I checked the permissions, and the file is readable.

**Comment by berndpfrommer** [comment]
Desperate digging around in the openhab sources to find a solution:
1) The Sonos binding uses Require-Bundle, but that is apparently no longer allowed.
2) The Daikin binding simply checked in the binary jar files into lib:
./lib/httpclient-4.3.3.jar
./lib/httpcore-4.3.2.jar
./lib/commons-logging-1.1.3.jar

I can do solution 2) but I'm not sure that's what you want. The two http*.jar files are 872kb combined, adding to the considerable weight of the source tree.

@teichsta, please advise.

**Comment by teichsta** [comment]
Hi @berndpfrommer, neither 1) nor 2) are good examples (will follow-up with the developers and hope to remove this dependencies as well). 

> I get this error when I try mvn clean install:
>[ERROR] import org.apache.http.impl.client.DefaultHttpClient;

that tells you, that have to add ```org.apache.http.impl.client``` to your imports as well.

Hope this helps,

Thomas E.-E.

**Comment by berndpfrommer** [comment]
I thought I had added all necessary imports, but apparently not so. Either way it compiles and runs now so I'm resubmitting this PR.
As you can tell I have tacked on some additional changes from @jrbenito that make PR #2120 obsolete.

**Comment by teichsta** [comment]
Thanks a lot @berndpfrommer!

**Pull request merged**